src/Common/src/System/Net/SocketAddress.cs

```diff
@@ -229,18 +229,50 @@ public override int GetHashCode()
 
         public override string ToString()
         {
-            var sb = new StringBuilder().Append(Family.ToString()).Append(':').Append(Size).Append(":{");
```

Comment:
In theory Roslyn could analyze all Append/AppendLine (+ escape analysis) and estimate a perfect initial capacity 🙂 Like you did: 
`sb.Append(someInteger)` // +10  (max length for a positive Int32)
`sb.Append("literal");` // + "literal".Length

Reply:
"perfect"... that would be really hard. For example, it would need to figure out that Size can't be negative.